app/src/main/java/fr/neamar/kiss/IconHandler.java

```diff
@@ -0,0 +1,133 @@
+package fr.neamar.kiss;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.util.Log;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+
+import fr.neamar.kiss.drawablecache.DiskDrawableCache;
+import fr.neamar.kiss.drawablecache.MemoryDrawableCache;
+
+
+public class IconHandler {
+
+    private static final String TAG = "IconHandler";
+    private final MemoryDrawableCache memoryCache = new MemoryDrawableCache();
+    private final DiskDrawableCache diskCache;
+    private IconPack iconPack;
+
+    public IconHandler(Context ctx) {
+        super();
+        iconPack = KissApplication.getIconPack();
+        diskCache = new DiskDrawableCache(ctx);
+    }
+
+    public void reset() {
+        iconPack = KissApplication.getIconPack();
+        memoryCache.clear();
+        diskCache.clear();
+    }
+
+    private String getCacheKey(ComponentName componentName) {
+        if (iconPack != null) {
+            return iconPack.packageName + "_" + componentName.toString();
+        }
+        return "default_" + componentName.toString();
```

Comment:
might be over-optimizing, but this string will be repeated a lot of time for every user not using a theme -- can we just use componentName() and skip the prefix?